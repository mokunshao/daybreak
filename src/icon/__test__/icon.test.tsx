import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Icon from '../icon';

describe('Icon', () => {
  it('是个 svg', () => {
    const { asFragment } = render(<Icon name="qq" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('onClick', () => {
    const fn = jest.fn();
    const { getByTestId } = render(<Icon name="qq" onClick={fn} data-testid="qq" />);
    fireEvent.click(
      getByTestId('qq'),
    );
    expect(fn).toBeCalled();
  });
});
