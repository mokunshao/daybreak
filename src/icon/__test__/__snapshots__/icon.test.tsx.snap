// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 是个 svg 1`] = `
<DocumentFragment>
  <svg
    class="daybreak-icon"
  >
    <use
      xlink:href="#qq"
    />
  </svg>
</DocumentFragment>
`;
